src/map.rs

```diff
@@ -1310,7 +1309,10 @@ where
 
                     next_table.store_bin(i, low_bin);
                     next_table.store_bin(i + n, high_bin);
-                    table.store_bin(i, Owned::new(BinEntry::Moved(next_table as *const _)));
+                    table.store_bin(
+                        i,
+                        table.get_moved(Shared::from(next_table as *const _), guard),
```

Comment:
Isn't `next_table` here _already_ a `Shared`?

Reply:
Here `next_table` is `deref`ed before. I moved this after the `if` in 1197 to avoid casting, but the immediately preceding operations take a reference. So either this or turn everything else into `unsafe { next_table.deref() }` inline I think